### PR TITLE
feat: NLP tool

### PR DESCRIPTION
**Pull request by lixinghua123**

### CODE REVIEW

File: packages/lb-components/src/components/NLPToolView/textContent/index.tsx [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+      const parentRect = contentRef.current.getBoundingClientRect();
+      const relativeLeft = elementRect.right - parentRect.left;
+      const relativeTop = elementRect.bottom - parentRect.top - element.offsetHeight;
+      if (relativeLeft && relativeTop) {
+        setRemarkStyle({ left: relativeLeft, top: relativeTop });
+      }
+    }
+  }, [remark?.editAuditID]);
+
+  const onSelectiRemark = (text: string) => {
```
Comment (MMMMoriaty) [inline packages/lb-components/src/components/NLPToolView/textContent/index.tsx:124]: wrong spell

File: packages/lb-components/src/components/NLPToolView/index.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    }
+    const currentData = imgList[imgIndex] ?? {};
+    const result = getCurrentResultFromResultList(currentData?.result);
+    const currentResult = result?.length > 0 ? result[0] : result;
+  }, [imgIndex]);
+
+  useEffect(() => {
+    if (stepList && step) {
+      const NLPStepConfig = getStepConfig(stepList, step)?.config;
+      setNLPConfig(jsonParser(NLPStepConfig));
```
Comment (Glenfiddish) [inline packages/lb-components/src/components/NLPToolView/index.tsx:103]: why not useMemo?

File: packages/lb-components/src/components/NLPToolView/textContent/index.tsx [hunk 3]
```diff
@@ -4,38 +4,101 @@
...
     if (lang) {
       i18n?.changeLanguage(lang);
     }
   }, []);
 
+  useEffect(() => {
+    onSelectionChange?.(selection.text)
+  }, [selection.text])
+
+  const renderContent = () => {
```
Comment (Glenfiddish) [inline packages/lb-components/src/components/NLPToolView/textContent/index.tsx:210]: It is recommended not to use render fn in hooks, and refactor into hooks

File: packages/lb-demo/package.json [hunk 4]
```diff
@@ -1,53 +1,53 @@
...
-    "development": [
-      "last 1 chrome version",
-      "last 1 firefox version",
-      "last 1 safari version"
-    ]
-  },
-  "devDependencies": {
-    "antd": "^4.16.13"
-  }
+	"name": "lb-demo",
```
Comment (Glenfiddish) [inline packages/lb-demo/package.json:2]: The indentation seems wrong, please check the autoformat on json

File: packages/lb-components/src/components/NLPToolView/index.tsx [hunk 5]
```diff
@@ -114,6 +114,16 @@ const NLPToolView: React.FC<IProps> = (props) => {
     updateSidebar()
   }
 
+  const clearResult = () => {
```
Comment (Glenfiddish) [inline packages/lb-components/src/components/NLPToolView/index.tsx:134]: Why is it not an empty list?

File: packages/lb-components/src/components/NLPToolView/textContent/index.tsx [hunk 6]
```diff
@@ -74,37 +92,180 @@ const TextContent: React.FC<IProps> = (props) => {
...
+          setRemarkResut({});
+          window.getSelection()?.empty();
+        },
+        submitData: remarkResut,
+      });
+    }
+    return null;
+  };
+
+  const renderRemarkMask = () => {
```
Comment (Glenfiddish) [inline packages/lb-components/src/components/NLPToolView/textContent/index.tsx:172]: render xx fn should split into hooks

File: packages/lb-components/src/views/MainView/sidebar/NLPSidebar/indicatorDetermine.tsx [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+  if (indicatorDetermine?.length > 0) {
+    return (
+      <div style={{ padding: '12px', marginBottom: '24px' }}>
+        <div style={{ fontSize: '16px', fontWeight: 500, lineHeight: '46px' }}>
+          {t('IndicatorJudgment')}
+        </div>
+        <div style={{ maxHeight: '400px', overflow: 'auto' }}>
+          {indicatorDetermine.map((item: IndicatorDetermine, index: number) => {
+            const { label, value } = item;
+            return label ? (
```
Comment (Glenfiddish) [inline packages/lb-components/src/views/MainView/sidebar/NLPSidebar/indicatorDetermine.tsx:52]: filter from list(indicatorDetermine) before it used

File: packages/lb-components/src/components/NLPToolView/index.tsx (hunk 5 above)
Comment (MMMMoriaty) [reply packages/lb-components/src/components/NLPToolView/index.tsx:134]: clearResult means textAnnotation is an empty list

**Pull request merged**